common/lib/xmodule/xmodule/util/adaptive_learning.py

```diff
@@ -53,136 +53,163 @@ def _adaptive_learning_configuration(self):
         )
 
     @lazy
-    def _adaptive_learning_url(self):
+    def _pending_reviews_url(self):
+        """
+        Return URL for accessing pending reviews.
+        """
+        instance_url = self._instance_url(self._adaptive_learning_configuration)
+        return '{base_url}/review_utils/fetch_reviews'.format(base_url=instance_url)
+
+    @lazy
+    def _request_headers(self):
+        """
+        Return custom headers for requests to external service that provides adaptive learning features.
+        """
+        return self.__request_headers(self._adaptive_learning_configuration)
+
+    @classmethod
+    def _adaptive_learning_url(cls, adaptive_learning_configuration):
```

Comment:
@pomegranited Turning the mixin into a class that can be instantiated on its own is a great idea, thanks :) I made that change. Just note that the new class no longer gets mixed in to `AdaptiveLibraryContentModule`: @e-kolpakov mentioned [here](https://github.com/open-craft/edx-platform/pull/49#discussion_r93747252) that

> there's an explicit instruction to never override `__init__` as it might break XBlock lifecycle

and that there is a chance that this might affect XModules as well. So instead of keeping the new class as a parent class for `AdaptiveLibraryContentModule` and giving it an `__init__` that takes an `adaptive_learning_configuration` or `course` parameter, I added an [`api_client` property](https://github.com/open-craft/edx-platform/pull/50/files#diff-5b223f1aa12d0cba135cf45dade27de3R401) to `AdaptiveLibraryContentModule` that instantiates the new class with the block's `parent_course`.